### DISCUSSION TITLE
SAMD21: are color BMP icons possible?

### DISCUSSION THREAD

**Discussion by deruiter** [Bodmer/TFT_eSPI#1674] [Q&A - General] [answered]
My system:
- Arduino Nano 33 IoT with SAMD21 processor.  
- Display: Adafruit 1.8" TFT breakout board with SD card slot - ST7735 controller

I love your amazing TFT library!   Busy with a project and I would like to display, say 30x30px, 256 color icons, reading them from the SD card.
Just want to know if this is possible so I can stop searching. 

I went through many examples and Google searches but I did not find if this is possible with my Arduino board??

Thank you!

I'm asking to use it for my Open Source Bathroom Fan Controller v2
hackster.io/edr1924

**Comment by Bodmer** [comment, marked as answer]
There is an example of BMP rendering here:
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Generic/TFT_SPIFFS_BMP

That one uses the SPIFFS filing system but it could be adapted, as a starting point you replace the word SPIFFS with SD, then solve any compile errors my looking at the SD library examples.
